File: t661/main.js
class Counter {
    constructor() {
        this.sum = 0;
        this.count = 0;
    }

    add(v) {
        this.sum += v;
        this.count += 1;
    }

    value() {
        return Math.floor(this.sum / this.count);
    }
}

const dir = [[-1,-1],[-1,0],[-1,1],
    [0,-1],[0,0],[0,1],
    [1,-1],[1,0],[1,1]];

/**
 * @param {number[][]} M
 * @return {number[][]}
 */
const imageSmoother = function(M) {
    const A = M.map(l => {
        return l.map(() => 0);
});
    for (let i = 0; i < M.length; i++) {
        for (let j = 0; j < M[i].length; j++) {
            const counter = new Counter();
            for (const d of dir) {
                const x = i + d[0];
                const y = j + d[1];
                if (x >= 0 && y >= 0 && x < M.length && y < M[i].length) {
                    counter.add(M[x][y]);
                }
            }
            A[i][j] = counter.value();
        }
    }
    return A;
};

File: t661/main.go
package main

import "fmt"

func imageSmoother(M [][]int) [][]int {
	h := len(M)
	A := make([][]int, h)

	for i := 0; i < h; i++ {
		w := len(M[i])
		Ai := make([]int, w)
		for j := 0; j < w; j++ {
			sum, cnt := 0, 0
			for px := -1; px < 2; px++ {
				for py := -1; py < 2; py++ {
					x := i + px
					y := j + py
					if x >= 0 && y >= 0 && x < h && y < w {
						sum += M[x][y]
						cnt += 1
					}
				}
			}
			Ai[j] = sum / cnt
		}
		A[i] = Ai
	}

	return A
}

func main() {
	fmt.Println(3 / 4)
}
